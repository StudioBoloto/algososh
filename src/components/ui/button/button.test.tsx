import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react';
import {Button} from './button';

describe('Button Component', () => {
    it('renders button with text correctly', () => {
        const {container} = render(<Button text="Click Me"/>);
        expect(container).toMatchSnapshot();
    });

    it('renders button without text correctly', () => {
        const {container} = render(<Button/>);
        expect(container).toMatchSnapshot();
    });

    it('renders disabled button correctly', () => {
        const {container} = render(<Button disabled/>);
        expect(container).toMatchSnapshot();
    });

    it('renders loading button correctly', () => {
        const {container} = render(<Button isLoader/>);
        expect(container).toMatchSnapshot();
    });

    it('calls onClick callback when button is clicked', () => {
        const onClickMock = jest.fn();
        render(<Button text="Click Me" onClick={onClickMock}/>);
        fireEvent.click(screen.getByText('Click Me'));
        expect(onClickMock).toHaveBeenCalledTimes(1);
    });
});
