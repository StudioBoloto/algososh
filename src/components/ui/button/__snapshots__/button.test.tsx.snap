// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders button with text correctly 1`] = `
<div>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    >
      Click Me
    </p>
  </button>
</div>
`;

exports[`Button Component renders button without text correctly 1`] = `
<div>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</div>
`;

exports[`Button Component renders disabled button correctly 1`] = `
<div>
  <button
    class="text text_type_button text_color_primary button undefined false "
    disabled=""
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</div>
`;

exports[`Button Component renders loading button correctly 1`] = `
<div>
  <button
    class="text text_type_button text_color_primary button undefined loader "
    disabled=""
    type="button"
  >
    <img
      alt="Загрузка."
      class="loader_icon"
      src="loader.svg"
    />
  </button>
</div>
`;
